src/js/legend.js

```diff
@@ -25,19 +25,61 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
             // dataSet accepts:
             // 1. an array of objects. Must contain "id", "value" and "label" variables.
             // Example: [{id: string, value: number, label: string} ... ]
-            dataSet: []
+            dataSet: [],
+            total: {
+                // value: number,
+                // percentage: number
+            }
         },
+        modelRelay: [{
```

Comment:
@cindyli, my one concern with this would be that approach seems to assume the pie and legend components will always be used in conjunction with a dataEntry component. I think each component needs to be able to independently calculate its totals and percentages. But I take the point about the repetition - is there a way we can avoid this while still keeping each component independently capable?